doc/nrf/app_dev/device_guides/nrf54l/ecies_x25519.rst

```diff
@@ -0,0 +1,61 @@
+.. _ecies_x25519:
+
+MCUboot AES Image Encryption with ECIES-X25519 Key Exchange
+###########################################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Introduction
+************
+
+MCUboot, when used with nRF54l15, can be configured to support encrypted images using AES.
+Images are encrypted using AES and ECIES-X25519 is used for key delivery (exchange) within image.
+When image encryption is enabled user may choose to either upload just signed or encrypted images to be swapped during boot.
+If MCUboot finds encrypted image in secondary slot it will decrypt it while swapping slots.
+Image that has been encrypted before being swapped into primary slot will be re-encrypted in case when swap is reverted.
+
+Limitations
+***********
+
+ECIES-X25519 key exchange is supported only when ED25519 signature is selected, which is default for nRF54l15.
+Encryption is not supported with MCUboot Direct-XIP modes.
+Storing ECIES-X25519 device private key in KMU is not supported at this moment.
+SHA256 is currently used by HMAC/HKDF, which is a subject to change.
+
+HMAC/HKDF impact on TLV and Key Exchange
+****************************************
+
+Encrypted image contains TLV that provides public key for ECIES-X25519 key exchange, encrypted AES key and MAC tag of the encrypted key.
+Derivation of AES key encryption key involves HKDF, and MAC tag, of encryption key, is generated using HMAC; both HKDF and HMAC use sha256, at this point.
+While using SHA256 is not a problem and does not significantly impact security it increases code size, as it requires addition of the algorithm support together with SHA512, which is already used by ED25519.
+There might be an option, in the future, to switch to using SHA512, but pre-installed MCUboot instance will not be able to boot images signed with TLV using SHA512.
```

Comment:
```suggestion
An encrypted image includes a TLV that contains the public key used for ECIES-X25519 key exchange, the encrypted AES key, and the MAC tag of the encrypted key.
The key used to encrypt the AES key is derived using HKDF, and the MAC tag is generated using HMAC.
Both HKDF and HMAC currently use the SHA-256 hash algorithm.

While the use of SHA-256 does not pose a security concern and has minimal impact on performance, it increases code size because SHA-256 support must be included alongside SHA-512, which is already used by the ED25519 signature algorithm.

Pre-installed MCUboot instances will not be able to boot images that use TLVs generated with different hash  algorithms that might be supported by this implementation in the future.
```